src/app/views/apps-management/apps-management.controller.coffee

```diff
@@ -1,29 +1,65 @@
 angular.module 'mnoEnterpriseAngular'
   .controller('AppsManagementCtrl',
-    ($q, $scope, MnoeConfig, MnoeProductInstances, MnoeProvisioning, MnoeOrganizations, AppManagementHelper) ->
+    ($q, $scope, MnoeConfig, MnoeProductInstances, MnoeProvisioning, MnoeOrganizations, AppManagementHelper, MnoeAppInstances) ->
 
       vm = @
       vm.isLoading = true
       vm.recentSubscription = AppManagementHelper.recentSubscription
 
+      getSyncStatusValue = (product) ->
+        sync_status = _.find(vm.connec_apps, (app) -> app.uid == product.uid)
+        return 'Disconnected' unless sync_status?.status
+
+        if sync_status.status.toLowerCase() in ['error', 'disconnected']
+          'Disconnected'
+        else
+          'Connected'
+
+      getSyncStatuses = ->
+        vm.products = _.map(vm.products,
+          (product) ->
+            if product.uid in vm.filterSyncProductIds
+              product.sync_status = {}
+              product.sync_status.attributes = {}
+              product.sync_status.attributes.status = getSyncStatusValue(product)
+            product
+        )
+
       vm.providesStatus = (product) ->
         vm.dataSharingEnabled(product) || product.subscription
 
       vm.dataSharingEnabled = (product) ->
         MnoeConfig.isDataSharingEnabled() && product.data_sharing
 
       vm.dataSharingStatus = (product) ->
-        if product.sync_status?.attributes?.status
-          'Connected'
+        # If sync status is set from backend then it follows following:
+        # sync_status values:
+        #   null => disconnected
+        #   all other values => connected
+        if _.find(vm.filterSyncProductIds, (uid) -> uid == product.uid)
+          product.sync_status?.attributes?.status
```

Comment:
Probably don't need the guarding here since if the product is in `vm.filterSyncProductIds` then it will always have the status set.